### PR TITLE
Proposal: updateTidbCluster not block when error occurs

### PR DESCRIPTION
**Pull request by mianhk**
### What problem does this PR solve?

Ref: #1245 
### Code changes

- [ ] Has Go code change
- [ ] Has CI related scripts change

### Tests

- [ ] Unit test 
- [ ] E2E test 
- [ ] Manual test 
- [ ] No code 

### Side effects

- [ ] Breaking backward compatibility 
- [ ] Other side effects: 

### Related changes

- [ ] Need to cherry-pick to the release branch 
- [ ] Need to update the documentation 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note
None
```

### CODE REVIEW

File: docs/design-proposals/2021-08-27-updateTidbCluster-no-blocking.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+- Add more e2e-test with syncing components error.
+- Test Upgrade components.Be not able to sync subsequent components when component is upgrading. 
+- Test Sync components with one component occurs error and other component success. Be able to sync subsequent components when component occurs error and is not upgrading.this may contains many cases:
+  1. With PD and TiKV Sepc changed.Sync PD error, sync TiKV success
+  2. With PD and TiKV TiDB changed.Sync PD error, sync TiDB success. 
+  3. With TiKV and TiDB Sepc changed.Sync TiKV success,sync TiDB success.
+  4. ...
+- Test if other components' Services/Statefulsets/PVCs... are normal or not when one component occurs error.  
+- Manual tests.
+####
```
Comment (tisonkun) [inline docs/design-proposals/2021-08-27-updateTidbCluster-no-blocking.md:39]:
```suggestion
```

Redundant.

Also could you try to reformat markdown with markdown-lint or at least add blank lines between items?

Comment (mianhk) [reply docs/design-proposals/2021-08-27-updateTidbCluster-no-blocking.md]: OK, fixed

Comment (tisonkun) [reply docs/design-proposals/2021-08-27-updateTidbCluster-no-blocking.md]: Thank you!

File: docs/design-proposals/2021-08-27-updateTidbCluster-no-blocking.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    errs = append(errs, err)
+}
+...
+return errorutils.NewAggregate(errs)
+```
+
+### Test Plan
+
+- Add more e2e-test with syncing components error.
+- Test Upgrade components.Be not able to sync subsequent components when component is upgrading.
```
Comment (csuzhangxc) [inline docs/design-proposals/2021-08-27-updateTidbCluster-no-blocking.md:42]:
```suggestion
- Test Upgrade components. Be not able to sync subsequent components when a component is upgrading. 
```

some more tiny grammatical issues

**Review by csuzhangxc — APPROVED** [review]

**Review by july2993 — APPROVED** [review]

**Pull request merged**